src/oracle/oracleDataController.ts

```diff
@@ -0,0 +1,39 @@
+import { Controller, Get, Route, Tags } from "tsoa";
+import { getLatestSubmission, getLatestSubmissionForEachOracle } from "./oracleDataService";
+import { getPolkaBtc } from "../common/polkaBtc";
+import { OracleStatus } from "./oracleModel";
+
+interface CachedOracleData {
+    maxDelay: number;
+    feed: string;
+    namesMap: Map<string, string>
+}
+
+@Tags("stats")
+@Route("oracle")
+export class OracleDataController extends Controller {
+    cachedOracleData: Promise<CachedOracleData> = (async () => {
+        const polkabtc = await getPolkaBtc();
+        const maxDelay = await polkabtc.oracle.getMaxDelay();
+        const feed = await polkabtc.oracle.getFeed();
+        const namesMap = await polkabtc.oracle.getSourcesById();
+        return {
+            maxDelay, feed, namesMap
+        };
+    })();
+
+    @Get("/submissions")
+    public async getLatestSubmissionForEachOracle(): Promise<OracleStatus[]> {
+        const oracleCache = await this.cachedOracleData;
+        const polkabtc = await getPolkaBtc();
```

Comment:
why do we need to submit the polkaBtc API here actually?

Reply:
By the way, just to clarify, `getPolkaBtc()` fetches a singleton instance of the lib and doesn't actually incur any external calls or delays.

Reply:
are you sure? Doesn't this connect to the chain? https://github.com/interlay/polkabtc-stats/blob/f04cecdff25297ce98bfdaf0f744b8ce91e3a371/src/common/polkaBtc.ts#L7

Reply:
It might, but a connection to the chain from stats is necessary anyway in order to retrieve some constants such as the BTC confirmations count or the oracles mapping. My point was rather that once we already have an instantiated `polkaBtc` instance, calling `getPolkaBtc()` again doesn't incur any extra costs, so I see no issue with using it here.

Reply:
But in this case, you are awaiting a new connection on every call to stats, right? So every new call to `submissions` makes a new connection to the parachain instead of using a "global" API object.

Reply:
I think instead of making the connection to the chain, you'd probably want to only have the types: https://github.com/interlay/polkabtc-js/blob/master/src/factory.ts#L36

Reply:
> But in this case, you are awaiting a new connection on every call to stats, right?
No, the connection instance is stored in memory
> So every new call to submissions makes a new connection to the parachain instead of using a "global" API object.
But a global API object is exactly what's being used. It's just wrapped in a promise to avoid race conditions on startup (i.e. trying to use the API before it's actually been initialised).

For this case Dan has already switched it to the type registry, but just wanted to clarify in general.